File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/published/bun/projects/l3-rewrite-conversions/converted.ts
import * as pulumi from "@pulumi/pulumi";
import * as primitive from "@pulumi/primitive";

interface ConvertedArgs {
    boolean: pulumi.Input<boolean>,
    float: pulumi.Input<number>,
    integer: pulumi.Input<number>,
    string: pulumi.Input<string>,
    numberArray: pulumi.Input<number[]>,
    booleanMap: pulumi.Input<Record<string, pulumi.Input<boolean>>>,
}

export class Converted extends pulumi.ComponentResource {
    constructor(name: string, args: ConvertedArgs, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:Converted", name, args, opts);
        const res = new primitive.Resource(`${name}-res`, {
            boolean: args.boolean,
            float: args.float,
            integer: args.integer,
            string: args.string,
            numberArray: args.numberArray,
            booleanMap: args.booleanMap,
        }, {
            parent: this,
        });

        this.registerOutputs();
    }
}

File: sdk/python/cmd/pulumi-language-python/testdata/setuppy/published/projects/l3-rewrite-conversions/converted.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import builtins as _builtins
import pulumi_primitive as primitive

class ConvertedArgs(TypedDict):
    boolean: Input[_builtins.bool]
    float: Input[_builtins.float]
    integer: Input[_builtins.int]
    string: Input[_builtins.str]
    numberArray: Input[list[_builtins.float]]
    booleanMap: Input[Dict[_builtins.str, _builtins.bool]]

class Converted(pulumi.ComponentResource):
    def __init__(self, name: str, args: ConvertedArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:Converted", name, args, opts)

        res = primitive.Resource(f"{name}-res",
            boolean=args["boolean"],
            float=args["float"],
            integer=args["integer"],
            string=args["string"],
            number_array=args["numberArray"],
            boolean_map=args["booleanMap"],
            opts = pulumi.ResourceOptions(parent=self))

        self.register_outputs({})
